actuators/machine.go

```diff
@@ -45,13 +47,21 @@ const (
 type Machine struct {
 	Core       corev1.CoreV1Interface
 	ClusterAPI v1alpha1.ClusterV1alpha1Interface
+	Log logr.Logger
+}
+
+type MachineParams struct {
+	Core       corev1.CoreV1Interface
+	ClusterAPI v1alpha1.ClusterV1alpha1Interface
+	LoggingContext string
 }
 
 // NewMachineActuator returns a new machine actuator object
-func NewMachineActuator(clusterapi v1alpha1.ClusterV1alpha1Interface, core corev1.CoreV1Interface) *Machine {
+func NewMachineActuator(params MachineParams) *Machine {
 	return &Machine{
-		Core:       core,
-		ClusterAPI: clusterapi,
+		Core:       params.Core,
+		ClusterAPI: params.ClusterAPI,
+		Log: klogr.New().WithName(params.LoggingContext),
```

Comment:
We shouldn't force klogr as the specific implementation here. I'm totally ok if the manager is forcing klogr, but the MachineActuator should only depend on the interface.

Reply:
So under `struct Machine` I put in `Log logr.Logger` which is an interface. Do you mean during the instantiation part in `NewMachineActuator(...)` to pass the logger type as an interface as well?

Reply:
i think so! 

basically I don't want klogr to be imported here, i want klogr to be imported in cmd/manager and keep any implementation of the logr.Logger interface out of the actuators package